### PR TITLE
smlm_datafusion3d app install

### PR DESCRIPTION
**Pull request by gyear**
For INC1315801 - `MATLAB-2022a.eb diplib-3.3.0-foss-2021a.eb Thrust-1.11.0-foss-2021a-CUDA-11.3.1.eb cub-1.11.0-foss-2021a-CUDA-11.3.1.eb smlm_datafusion3d-v1.0-alpha-foss-2021a-CUDA-11.3.1.eb`

* [x] Assigned to reviewers (usually everyone in apps team)

Default:
* [ ] EL8-icelake
* [ ] EL8-haswell

**Note** Matlab needs to go throught hte same proceedure as here: <https://bearwiki.bham.ac.uk/bear/doku.php?id=arc:rsg:bear-apps:building:app-specific:matlab#file_installation_keys> 
**Apps needs to be built on the GPUs** (CUDA extension)

### CODE REVIEW

File: easybuild/easyconfigs/t/Thrust/Thrust-1.11.0-foss-2021a-CUDA-11.3.1.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    ('CUDA', '11.3.1', '', True),
+]
+
+separate_build_dir = False
+buildininstalldir = True
+
+# sed command based on line position in CMakeLists.txt
+preconfigopts = 'sed -i 46s/\"ON\"/\"OFF\"/ CMakeLists.txt && '
+preconfigopts += 'sed -i 47s/\"ON\"/\"OFF\"/ CMakeLists.txt && '
+preconfigopts += 'sed -i 48s/\"ON\"/\"OFF\"/ CMakeLists.txt && '
```
Comment (branfosj) [inline easybuild/easyconfigs/t/Thrust/Thrust-1.11.0-foss-2021a-CUDA-11.3.1.eb]: Does the standard `configopts = "-Dbar=OFF` format not work here?

If not please add to the comment to explain what options we we are setting to OFF.

File: easybuild/easyconfigs/s/smlm_datafusion3d/smlm_datafusion3d-v1.0-alpha-foss-2021a-CUDA-11.3.1.eb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+buildininstalldir = True
+
+skipsteps = ['install']
+
+configopts = "-DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_C_COMPILER=cc -DCUB_ROOT_DIR=$EBROOTCUB/include"
+
+# sed command based on line position in CMakeLists.txt
+prebuildopts = 'sed -i 57s/\"3.0\"//g ../smlm_datafusion3d/CMakeLists.txt && '
+prebuildopts += 'sed -i 57s/\"3.5\"//g ../smlm_datafusion3d/CMakeLists.txt && '
+prebuildopts += 'sed -i "62d;63d;64d" ../smlm_datafusion3d/CMakeLists.txt && '
```
Comment (branfosj) [inline easybuild/easyconfigs/s/smlm_datafusion3d/smlm_datafusion3d-v1.0-alpha-foss-2021a-CUDA-11.3.1.eb]: Add to the comment to specify that you are changing the CUDA architectures to those supported by the CUDA version.

**Review by branfosj — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/smlm_datafusion3d/smlm_datafusion3d-v1.0-alpha-foss-2021a-CUDA-11.3.1.eb (hunk 2 above)
Comment (gyear) [reply easybuild/easyconfigs/s/smlm_datafusion3d/smlm_datafusion3d-v1.0-alpha-foss-2021a-CUDA-11.3.1.eb]: MR to add detail

File: easybuild/easyconfigs/t/Thrust/Thrust-1.11.0-foss-2021a-CUDA-11.3.1.eb (hunk 1 above)
Comment (gyear) [reply easybuild/easyconfigs/t/Thrust/Thrust-1.11.0-foss-2021a-CUDA-11.3.1.eb]: removed as Thrust seems to work without this

**Review by branfosj — APPROVED** [review]

**Pull request merged**